src/Containers/OhmmsPETE/OhmmsVector.h

```diff
@@ -301,6 +290,34 @@ class Vector
     nLocal     = n;
     nAllocated = n;
   }
+
+  inline static void construct_fill_elements(Type_t* ptr, size_t n_elements, const Type_t& val)
+  {
+    if constexpr (std::is_trivial<T>::value)
+      qmc_allocator_traits<Alloc>::fill_n(ptr, n_elements, val);
+    else if constexpr (qmc_allocator_traits<Alloc>::is_host_accessible)
+      for (size_t i = 0; i < n_elements; i++)
+        new (ptr + i) Type_t(val);
+  }
+
+  inline static void construct_copy_elements(const Type_t* from, size_t n_elements, Type_t* to)
+  {
+    if constexpr (qmc_allocator_traits<Alloc>::is_host_accessible)
+    {
+      if constexpr (std::is_trivial<T>::value)
+        std::copy_n(from, n_elements, to);
+      else
+        for (size_t i = 0; i < n_elements; i++)
+          new (to + i) Type_t(*(from + i));
+    }
+  }
+
```

Comment:
There was once the assumption you could marshal OhmmsContainers over MPI but just copying the bytes in X. Or attaching X to a part of a big buffer and then sending that buffer.  You just can't do that even with this more correct code.  I think the answer to that is just code review to keep such schemes from returning.

Reply:
I think you can still do that with MPI as long as you guarantee type safety. I don't get you mean here.

Reply:
Correct. Only non pointer value can be sent via MPI,